GREYErrorFormatter.m

```diff
@@ -0,0 +1,120 @@
+//
+//  GREYErrorFormatter.m
+//  CommonLib
+//
+//  Created by Will Said on 6/1/20.
+//
+
+#import "GREYErrorFormatter.h"
+
+#import "GREYObjectFormatter.h"
+#import "GREYInteraction.h"
+#import "GREYFatalAsserts.h"
+#import "GREYErrorConstants.h"
+
+@interface GREYErrorFormatter ()
+
+@property(readonly, nonatomic) GREYError *error;
+
+@end
+
+@implementation GREYErrorFormatter
+
+#pragma mark - Class Methods
+
++ (NSString *)formattedHierarchy:(nonnull NSString *)hierarchy {
+    NSMutableArray<NSString*> *logger = [[NSMutableArray alloc] init];
+    
+    [logger addObject:@"UI Hierarchy (ordered by window level, back to front):\n"];
+
+    NSString *windowLegend = @"[Window 1]";
+    NSString *axLegend = @"[AX]";
+    NSString *uieLegend = @"[UIE]";
+
+    NSDictionary<NSString *, NSString *> *legendLabels = @{
+      windowLegend : @"Back-Most Window",
+      axLegend : @"Accessibility",
+      uieLegend : @"User Interaction Enabled"
+    };
+    NSArray<NSString *> *keyOrder = @[ windowLegend, axLegend, uieLegend ];
+
+    NSString *legendDescription = [GREYObjectFormatter formatDictionary:legendLabels
+                                                                 indent:kGREYObjectFormatIndent
+                                                              hideEmpty:NO
+                                                               keyOrder:keyOrder];
+    [logger addObject:[NSString stringWithFormat:@"%@: %@\n", @"Legend", legendDescription]];
+    [logger addObject:hierarchy];
+    return [logger componentsJoinedByString:@"\n"];
+}
+
+#pragma mark - Init
+
+- (instancetype)initWithError:(GREYError *)error {
+  self = [super init];
+  if (self) {
+    _error = error;
+  }
+  return self;
+}
+
+#pragma mark - Public Methods
+
+- (NSString *)formattedDescription {
+  if ([_error.domain isEqualToString:kGREYInteractionErrorDomain] &&
+      _error.code == kGREYInteractionElementNotFoundErrorCode) {
```

Comment:
This is replicated in GREYError, you can just keep this here and return nil instead of crashing.